sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -156,9 +156,17 @@ object RapidsPluginUtils extends Logging {
       // Derive it from spark.executor.cores, since spark.executor.cores is not set on all cluster
       // managers by default, we should judge whether if it's set explicitly.
       if (conf.contains(EXECUTOR_CORES_KEY)) {
-        conf.set(numThreadsKey, Math.max(20, conf.get(EXECUTOR_CORES_KEY).toInt).toString)
+        conf.set(numThreadsKey, Math.max(RapidsConf.MULTITHREAD_READ_NUM_THREADS_DEFAULT,
+          conf.get(EXECUTOR_CORES_KEY).toInt).toString)
+        logWarning(s"$numThreadsKey is set to " + conf.get(numThreadsKey) + ".")
       }
     }
+
+    // If spark.rapids.sql.multiThreadedRead.numThreads is set explicitly as 0, and it can not
+    // be derived from other settings, we should throw an exception.
+    if (conf.contains(numThreadsKey) && conf.get(numThreadsKey).toInt == 0) {
+      throw new IllegalArgumentException(s"$numThreadsKey can not be zero.")
```

Comment:
I'm a bit confused by this check, @abellina had suggested moving it to the RapidsConf definition but looking it it looks like its already there:

```
val MULTITHREAD_READ_NUM_THREADS = conf("spark.rapids.sql.multiThreadedRead.numThreads")
      .doc("The maximum number of threads on each executor to use for reading small " +
        "files in parallel. This can not be changed at runtime after the executor has " +
        "started. Used with COALESCING and MULTITHREADED readers, see " +
        "spark.rapids.sql.format.parquet.reader.type, " +
        "spark.rapids.sql.format.orc.reader.type, or " +
        "spark.rapids.sql.format.avro.reader.type for a discussion of reader types.")
      .integerConf
      .checkValue(v => v > 0, "The thread count must be greater than zero.")
      .createWithDefault(20)
```

so I think just remove the checks for ==0 here and in the above check